### PR TITLE
[Skill System][Dependencies] Add typed skill dependency remediation

### PR DESCRIPTION
**Pull request by realmorrisliu**
## Summary
- add stable typed dependency support under `compatibility.dependencies`
- canonicalize `required_tools` and recognized compatibility hints into one availability/remediation model
- surface typed dependency diagnostics in runtime, CLI, and spec docs

## Testing
- cargo test -p alan-runtime
- cargo test -p alan --test skills_cli_integration_test
- cargo test -p alan skill_catalog
- cargo clippy -p alan-runtime --all-targets -- -D warnings
- cargo clippy -p alan --all-targets -- -D warnings

Refs #210

### CODE REVIEW

File: crates/runtime/src/skills/types.rs [hunk 1]
```diff
@@ -1477,22 +1562,142 @@ pub fn validate_capabilities(cap: &SkillCapabilities) -> Result<(), SkillsError>
...
+            validate_non_empty_dependency_name("dependency", name)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn validate_tool_name(name: &str) -> Result<(), SkillsError> {
+    if name.contains(' ') || name.contains('<') || name.contains('>') {
+        return Err(SkillsError::InvalidCapabilities(format!(
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply crates/runtime/src/skills/types.rs]: Fixed in `52ab943`.

`validate_tool_name(...)` now shares the same non-empty / no-whitespace guard as the other typed dependency validators before applying the existing tool-name-specific character checks. That closes the blank-string and tab/newline cases for both `compatibility.dependencies: [{ type: tool, ... }]` and legacy `required_tools` / `optional_tools` capability lists, so malformed skill metadata now fails fast during load instead of producing a confusing `tool:` availability gate at runtime.

File: crates/runtime/src/skills/types.rs [hunk 2]
```diff
@@ -821,6 +863,30 @@ impl SkillHostCapabilities {
...
+        I: IntoIterator<Item = S>,
+        S: Into<String>,
+    {
+        self.env_vars.extend(env_vars.into_iter().map(Into::into));
+    }
+
+    pub fn with_process_env(mut self) -> Self {
+        self.extend_env_vars(
+            std::env::vars_os().map(|(name, _)| name.to_string_lossy().into_owned()),
+        );
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply crates/runtime/src/skills/types.rs]: Fixed in `ace5b79`.

The runtime-facing change is now in `SkillHostCapabilities::with_process_env()`: it only records environment variables whose current value is non-empty, so an empty `OPENAI_API_KEY=""` no longer satisfies an `env_var` dependency. I kept the existing `with_env_vars(...)` helper semantics for tests and synthetic host-capability construction, but the production builders in `runtime/engine.rs`, `runtime/submission_handlers.rs`, and `alan/src/skill_catalog.rs` all flow through `with_process_env()`, so real availability/remediation now matches runtime behavior.

I also added a regression test covering the empty-value case and reran `cargo fmt --all`, `cargo test -p alan-runtime`, and `cargo clippy -p alan-runtime --all-targets -- -D warnings`.

File: crates/runtime/src/skills/types.rs [hunk 3]
```diff
@@ -832,6 +918,17 @@ impl SkillHostCapabilities {
         }
     }
 
+    pub fn supports_env_var(&self, name: &str) -> bool {
+        self.env_vars.contains(name)
+    }
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply crates/runtime/src/skills/types.rs]: Fixed in `bdfb509`.

Env-var dependency matching now normalizes names on both insertion and lookup using host-aware casing rules: on Windows we fold to uppercase, while Unix-like hosts preserve exact casing. That keeps Unix semantics intact but avoids false missing-dependency reports on Windows when the process exports a variable under a different case (for example `openai_api_key` vs `OPENAI_API_KEY`).

I applied the normalization consistently across `with_env_vars(...)`, `with_process_env()`, and `supports_env_var(...)`, added a regression test for the case-insensitive path, and reran `cargo fmt --all`, `cargo test -p alan-runtime`, and `cargo clippy -p alan-runtime --all-targets -- -D warnings`.
